megatron/training.py

```diff
@@ -1033,3 +1080,11 @@ def evaluate_and_print_results(
     print_rank_0("-" * length)
     print_rank_0(string)
     print_rank_0("-" * length)
+
+def save_snapshot(neox_args):
+    snapshot = torch.cuda.memory._snapshot()
+    snapshot_path = os.path.join(neox_args.memory_profiling_path)
```

Comment:
add an assert to make sure `memory_profiling_path` is not `None`, which is the arg default

Reply:
added